cpp/src/io/json/read_json.cu

```diff
@@ -109,45 +126,46 @@ rmm::device_uvector<char> ingest_raw_input(host_span<std::unique_ptr<datasource>
                       delimiter_source,
                       delimiter_source + d_delimiter_map.size(),
                       d_delimiter_map.data(),
-                      d_buffer.data());
+                      buffer.data());
     }
-
     stream.synchronize();
-    return d_buffer;
-
-  } else {
-    auto buffer = std::vector<uint8_t>(total_source_size);
-    // Single read because only a single compressed source is supported
-    // Reading to host because decompression of a single block is much faster on the CPU
-    sources[0]->host_read(range_offset, total_source_size, buffer.data());
-    auto const uncomp_data = decompress(compression, buffer);
-    return cudf::detail::make_device_uvector_sync(
-      host_span<char const>{reinterpret_cast<char const*>(uncomp_data.data()), uncomp_data.size()},
-      stream,
-      rmm::mr::get_current_device_resource());
+    return buffer.first(bytes_read);
   }
+  /* TODO: allow byte range reading from multiple compressed files.
+   */
```

Comment:
Nitpick:  Is there any chance we could consider using line comments instead of block comment?